subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_spi_stm32.c

```diff
@@ -745,6 +777,73 @@ struct ec_host_cmd_backend *ec_host_cmd_backend_get_spi(struct gpio_dt_spec *cs)
 	return &ec_host_cmd_spi;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int ec_host_cmd_spi_stm32_pm_action(const struct device *dev,
+					   enum pm_device_action action)
+{
+	const struct ec_host_cmd_backend *backend = (struct ec_host_cmd_backend *)dev->data;
+	struct ec_host_cmd_spi_ctx *hc_spi = (struct ec_host_cmd_spi_ctx *)backend->ctx;
+	const struct ec_host_cmd_spi_cfg *cfg = hc_spi->spi_config;
+	int err;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		/* Set pins to active state */
+		err = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+		if (err < 0) {
+			return err;
+		}
+
+		/* Enable device clock */
+		err = clock_control_on(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+				       (clock_control_subsys_t)&cfg->pclken[0]);
+		if (err < 0) {
+			return err;
+		}
+		/* Enable CS interrupts. */
+		gpio_pin_interrupt_configure_dt(&hc_spi->cs, GPIO_INT_EDGE_BOTH);
+
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+#ifdef SPI_SR_BSY
+		/* Prevent corruption of the last transfer */
+		while (LL_SPI_IsActiveFlag_BSY(cfg->spi) == 1) {
+		}
```

Comment:
would it be a good idea to bound this one and return an error if it timesout? You could use `WAIT_FOR`

Reply:
Nice suggestion. I think returning an error is not necessary here, because Host Command backend can be suspended in 2 cases: AP is going down or another backend is used. IMO the BSY flag shouldn't stop suspension, because of that. What do you think?